packages/mdc-dialog/index.ts

```diff
@@ -70,7 +70,7 @@ class MDCDialog extends MDCComponent<MDCDialogFoundation> {
   private container_!: HTMLElement; // assigned in initialize()
   private content_!: HTMLElement | null; // assigned in initialize()
   private defaultButton_!: HTMLElement | null; // assigned in initialize()
-  private initialFocusEl_!: HTMLElement | null; // assigned in initialize()
+  private initialFocusEl_!: HTMLElement | undefined; // assigned in initialize()
```

Comment:
This seems like a contradiction in terms. Isn't this effectively `private initialFocusEl_?: HTMLElement`?

Reply:
Confusingly, the `!` here has a slightly different meaning than when it's used after an expression.

When declaring member properties, `!` is a "[definite assignment assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#strict-class-initialization)", meaning "I pinky swear that _some_ value is assigned to this property, but it happens somewhere other than the constructor.

In an expression (e.g., `querySelector('.foo')!`), the bang means "this value will never be null or undefined."

You make a good point though; this can be shortened to `private initialFocusEl_?: HTMLElement`.

Reply:
Done